### ISSUE TITLE
typescript errors - cannot be used as a JSX component

### ISSUE THREAD

**Issue by LukasMod** [Abhinandan-Kushwaha/react-native-gifted-charts#269]
**Problem:**
After adding library to project and `yarn tsc -p ./tsconfig.json` i get:

```
Errors  Files
     9  node_modules/react-native-gifted-charts/src/BarChart/Animated2DWithGradient.tsx:119
    43  node_modules/react-native-gifted-charts/src/BarChart/index.tsx:434
    11  node_modules/react-native-gifted-charts/src/BarChart/RenderBars.tsx:142
    11  node_modules/react-native-gifted-charts/src/BarChart/RenderStackBars.tsx:109
    14  node_modules/react-native-gifted-charts/src/Components/AnimatedBar/index.tsx:49
     6  node_modules/react-native-gifted-charts/src/Components/lineSvg.tsx:22
    14  node_modules/react-native-gifted-charts/src/Components/ThreeDBar/index.tsx:39
    60  node_modules/react-native-gifted-charts/src/LineChart/index.tsx:2120
    49  node_modules/react-native-gifted-charts/src/LineChart/LineChartBicolor.tsx:848
     3  node_modules/react-native-gifted-charts/src/PieChart/index.tsx:96
    13  node_modules/react-native-gifted-charts/src/PieChart/main.tsx:193
```

which is mostly something like this:
`node_modules/react-native-gifted-charts/src/PieChart/main.tsx:449:10 - error TS2786: 'View' cannot be used as a JSX component.
  Its instance type 'View' is not a valid JSX element.`

tsconfig:

```
{
  "compilerOptions": {
    "allowJs": false,
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "jsx": "react-native",
    "module": "es2015",
    "moduleResolution": "node",
    "noImplicitAny": false,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noUnusedLocals": true,
    "target": "esnext",
    "lib": ["esnext", "dom"],
    "skipLibCheck": true,
    "resolveJsonModule": true
  },
  "exclude": ["node_modules"]
}
```
package.json
```
  "dependencies": {
    "expo": "~46.0.9",
    "expo-status-bar": "~1.4.0",
    "react": "18.0.0",
    "react-dom": "18.0.0",
    "react-native": "0.69.5",
    "react-native-gifted-charts": "^1.2.41",
    "react-native-linear-gradient": "^2.6.2",
    "react-native-svg": "^13.2.0",
    "react-native-web": "~0.18.7"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@types/react": "~18.0.14",
    "@types/react-native": "~0.69.1",
    "typescript": "~4.3.5"
  },
}
```

**Steps to reproduce:**
1. npx create-expo-app -t expo-template-blank-typescript
2. yarn add react-native-gifted-charts react-native-linear-gradient react-native-svg
3. add tsconfig.json and import inside App.tsx demo:
```
import { View } from "react-native"
import { BarChart, LineChart, PieChart } from "react-native-gifted-charts"

export default function App() {
  const data = [{ value: 50 }, { value: 80 }, { value: 90 }, { value: 70 }]

  return (
    <View>
      <BarChart data={data} />
      <LineChart data={data} />
      <PieChart data={data} />
    </View>
  )
}
```
4. yarn tsc -p ./tsconfig.json

I've already searched for solution and tried with different versions of typescript, @types/react, @types/react-native. Don't know why it is looking in node_modules even it's inside "exclude": ["node_modules"].

**Comment by Fischaela** [comment]
I have exactly the same problem. @LukasMod Did you find a solution?

**Comment by LukasMod** [comment]
Unfortunately not. I have currently changed the project so I don't have this problem anymore.

**Comment by otaviogaiao** [comment]
I'm having the same issue. It's causing dozens of typescript errors in my project. And  "exclude": ["node_modules"] doesn't work.

**Comment by radekzz** [comment]
I used [patch-package](https://www.npmjs.com/package/patch-package)

to insert
`//@ts-nocheck`
in the first line of every problematic file

**Comment by michaelpomogajko** [comment]
Any updates here? Also having the issue that my pipeline breaks due to typescript issue in this library

**Comment by arunpugahz** [comment]
Getting error at `tsc --noEmit --incremental false` command

Errors  Files
     1  node_modules/react-native-gifted-charts/src/BarChart/RenderBars.tsx:7
     1  node_modules/react-native-gifted-charts/src/BarChart/RenderStackBars.tsx:14
    10  node_modules/react-native-gifted-charts/src/Components/AnimatedBar/index.tsx:138
     1  node_modules/react-native-gifted-charts/src/Components/BarAndLineChartsWrapper/index.tsx:321

**Comment by Abhinandan-Kushwaha** [comment]
Hi @arunpugahz 👋
Thanks for pointing out specific typescript issues. They have been fixed in version `1.3.11` Please update to this version.

**Comment by samobrien13** [comment]
```
Found 98 errors in 12 files.

Errors  Files
     6  node_modules/react-native-gifted-charts/src/BarChart/index.tsx:76
     1  node_modules/react-native-gifted-charts/src/Components/BarAndLineChartsWrapper/index.tsx:322
     4  node_modules/react-native-gifted-charts/src/Components/BarAndLineChartsWrapper/renderHorizSections.tsx:350
     4  node_modules/react-native-gifted-charts/src/Components/BarAndLineChartsWrapper/renderLineInBarChart.tsx:7
     2  node_modules/react-native-gifted-charts/src/Components/BarAndLineChartsWrapper/renderVerticalLines.tsx:6
     1  node_modules/react-native-gifted-charts/src/Components/BarSpecificComponents/barBackgroundPattern.tsx:4
     1  node_modules/react-native-gifted-charts/src/Components/BarSpecificComponents/cap.tsx:5
     1  node_modules/react-native-gifted-charts/src/Components/common/Pointer.tsx:4
     1  node_modules/react-native-gifted-charts/src/Components/common/StripAndLabel.tsx:5
    46  node_modules/react-native-gifted-charts/src/LineChart/index.tsx:124
    23  node_modules/react-native-gifted-charts/src/LineChart/LineChartBicolor.tsx:291
     8  node_modules/react-native-gifted-charts/src/utils/index.tsx:69
```

Still seeing quite a few typescript errors
